x-pack/plugins/security_solution/server/endpoint/lib/files/check_deleted_files_task.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { CoreSetup, Logger } from '@kbn/core/server';
+import type {
+  ConcreteTaskInstance,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import { throwUnrecoverableError } from '@kbn/task-manager-plugin/server';
+
+import type { EndpointAppContext } from '../../types';
+import {
+  fileIdsWithoutChunks,
+  getFilesByStatus,
+  updateFilesStatus,
+} from '../../services/actions/action_files';
+
+export const TYPE = 'endpoint:check-deleted-files-task';
+export const VERSION = '1.0.0';
+const TITLE = 'Security Solution Endpoint Deleted Files Periodic Tasks';
+const TIMEOUT = '2m';
+const SCOPE = ['securitySolution'];
+const INTERVAL = '5m';
```

Comment:
how important is it to keep the `file.Status` up to date? should we reduce the internal?

Reply:
I think we account for the possible time gap latency in our UI (cc @paul-tavares). Not sure if Fleet team will also. That's probably the only effect. Otherwise it's to make sure the data stays "in-spec" and the File plugin plays nicely with everything into the future

Reply:
There is still the possibility that the UI might show a download link and the API returns an empty file/error, but that is an edge case that given the amount of time I think the Fleet team is looking to use in the clean up (like >30 days) the likely hood of that happening is probably very low.

Reply:
Actually - I take that back @pzl .
The UI does not have code to account for the possibility of this still happening - but the file info API does. It checks every file that has a status of `READY` to ensure that they have content (chunks) and if not, then it changes the `status` that is returned from that API to `DELETED`. So I think we're ok with the interval being `5m`... but now that I mention that, should it actually be larger interval?

Reply:
> So I think we're ok with the interval being 5m... but now that I mention that, should it actually be larger interval?

I was thinking this too, we could probably make the interval larger.  I was thinking maybe just every 24hrs.  My reasoning is that these will be files that were retrieved 30 days ago or more, so it probably won't be too often that users go back to an action that old to download an old file.  We could consider the primary use case for this is to be part of the clean up of older files and less of something users will encounter too often.

Thoughts?

Reply:
I changed it to 1 day in the fleet PR: https://github.com/elastic/kibana/pull/144494.